cnd/src/swap_protocols/rfc003/mod.rs

```diff
@@ -42,3 +44,46 @@ pub enum Error {
     #[error("internal error: {0}")]
     Internal(String),
 }
+
+/// Both Alice and Bob use `DeriveIdentities` together with:
+/// - the current cnd seed when requesting and accepting/declining a swap
+/// - the seed in the state store for an ongoing swap
+pub trait DeriveIdentities: Send + Sync + 'static {
+    fn derive_redeem_identity(&self) -> SecretKey;
+    fn derive_refund_identity(&self) -> SecretKey;
+}
+
+impl DeriveIdentities for Seed {
+    fn derive_redeem_identity(&self) -> SecretKey {
+        SecretKey::from_slice(self.sha256_with_seed(&[b"REDEEM"]).as_ref())
+            .expect("The probability of this happening is < 1 in 2^120")
+    }
+
+    fn derive_refund_identity(&self) -> SecretKey {
+        SecretKey::from_slice(self.sha256_with_seed(&[b"REFUND"]).as_ref())
+            .expect("The probability of this happening is < 1 in 2^120")
+    }
+}
+
+/// SecretHasher generates the secret from a hash of the current seed for
+/// inclusion in HTLC.  This 'secret' is what is revealed to Bob when Alice
+/// redeems the HTLC on beta ledger.
+pub trait SecretHasher {
```

Comment:
Do we really need this trait?

I'd prefer if we find a model that doesn't rely on a concept like this because it doesn't exist in our domain.